src/types/video.ts

```diff
@@ -177,7 +156,7 @@ export enum PosterResizeModeType {
 export type AudioOutput = 'speaker' | 'earpiece';
 
 export interface ReactVideoProps extends ReactVideoEvents, ViewProps {
-  source?: ReactVideoSource;
+  source: ReactVideoSource;
```

Comment:
source can be undefined. It is used to stop the player without releasing it. you should keep ? or put | undefined

Reply:
Fixed on this commit
https://github.com/react-native-video/react-native-video/pull/3487/commits/7944195e9dabcc6d4d9d6c7519903742ccb8df80